### PR TITLE
Full path to certificate for windows container

### PR DESCRIPTION
**Pull request by evgenykorolyuk**

### CODE REVIEW

File: aspnetcore/security/docker-https.md [hunk 1]
```diff
@@ -109,7 +109,7 @@ Run the container image with ASP.NET Core configured for HTTPS:
 
 ```console
 docker pull mcr.microsoft.com/dotnet/core/samples:aspnetapp
-docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
+docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-https.md:112]:
```suggestion
docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="$CREDENTIAL_PLACEHOLDER$" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
```

**Note:*** `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password.

**Comment by Rick-Anderson** [comment]
@richlander @mthalman @albertodenatale @pathanlan @plooploops 30 second review on changing `Path=\https\aspnetapp.pfx`  to `Path=c:\https\aspnetapp.pfx`

File: aspnetcore/security/docker-https.md [hunk 2]
```diff
@@ -109,9 +109,11 @@ Run the container image with ASP.NET Core configured for HTTPS:
 
 ```console
 docker pull mcr.microsoft.com/dotnet/core/samples:aspnetapp
-docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
+docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="$CREDENTIAL_PLACEHOLDER$" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
 ```
 
+***NOTE:*** `$CREDENTIAL_PLACEHOLDER$`is a placeholder for the Kestrel certificates default password.
```
Comment (mthalman) [inline aspnetcore/security/docker-https.md:115]: This line uses `$CREDENTIAL_PLACEHOLDER$` as the placeholder. But up on line 55, it uses `<password>`. And on line 122, it uses `{ password here }`. This all needs to get cleaned up to be consistent. The same syntax and placeholder name should be used in all cases. I prefer the angle bracket syntax for placeholders.

There are two other usages of `ASPNETCORE_Kestrel__Certificates__Default__Password="password"` that need to be updated as well.

File: aspnetcore/security/docker-https.md [hunk 3]
```diff
@@ -109,9 +109,11 @@ Run the container image with ASP.NET Core configured for HTTPS:
 
 ```console
 docker pull mcr.microsoft.com/dotnet/core/samples:aspnetapp
-docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
+docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="$CREDENTIAL_PLACEHOLDER$" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
```
Comment (plooploops) [inline aspnetcore/security/docker-https.md:112]: Nice catch on the environment variable value for the Windows using Windows Containers example:

ASPNETCORE_Kestrel__Certificates__Default__Path=`c:`\https\aspnetapp.pfx

**Comment by plooploops** [comment]
> @richlander @mthalman @albertodenatale @pathanlan @plooploops 30 second review on changing `Path=\https\aspnetapp.pfx` to `Path=c:\https\aspnetapp.pfx`

Based on local testing with Windows using Windows Containers, setting `ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx` works with the sample container.

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27186) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27186) before we can accept your contribution.<br/><br/>:x: evgenykorolyuk [sign now](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27186)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/AspNetCore.Docs?pullRequest=27186) it.</sub>

**Comment by Rick-Anderson** [comment]
@evgenykorolyuk sorry, my changes pushed this PR over the limit for not signing the CLA. Can you sign it?

File: aspnetcore/security/docker-https.md [hunk 4]
```diff
@@ -52,23 +52,22 @@ Use the following instructions for your operating system configuration.
 Generate certificate and configure local machine:
 
 ```dotnetcli
-dotnet dev-certs https -ep %USERPROFILE%\.aspnet\https\aspnetapp.pfx -p <password>
+dotnet dev-certs https -ep %USERPROFILE%\.aspnet\https\aspnetapp.pfx -p <$CREDENTIAL_PLACEHOLDER$>
```
Comment (mthalman) [inline aspnetcore/security/docker-https.md:55]: Can we get rid of the dollar signs? They don't add any value and could be confusing to readers. `<CREDENTIAL_PLACEHOLDER>`

Comment (Rick-Anderson) [reply aspnetcore/security/docker-https.md]: > Can we get rid of the dollar signs? They don't add any value and could be confusing to readers. `<CREDENTIAL_PLACEHOLDER>`

Done, See [this commit](https://github.com/dotnet/AspNetCore.Docs/pull/27186/commits/2021e914e151fc1211305fb5b2cef5277e4edd6a)

File: aspnetcore/security/docker-https.md [hunk 5]
```diff
@@ -76,41 +76,43 @@ Note: The certificate in this case must be a `.pfx` file.  Utilizing a `.crt` or
...
+In the preceding commands, replace `<CREDENTIAL_PLACEHOLDER>` with a password. When using [PowerShell](/powershell/scripting/overview), replace `%USERPROFILE%` with `$env:USERPROFILE`.
 
 Run the container image with ASP.NET Core configured for HTTPS:
 
 ```console
-docker pull mcr.microsoft.com/dotnet/samples:aspnetapp
-docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ --user ContainerAdministrator mcr.microsoft.com/dotnet/samples:aspnetapp
+docker pull mcr.microsoft.com/dotnet/core/samples:aspnetapp
+docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="CREDENTIAL_PLACEHOLDER" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-https.md:111]:
```suggestion
docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="CREDENTIAL_PLACEHOLDER" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ --user ContainerAdministrator mcr.microsoft.com/dotnet/core/samples:aspnetapp
```

Comment (Rick-Anderson) [reply aspnetcore/security/docker-https.md:111]: @mthalman if this is correct, please commit.

File: aspnetcore/security/docker-https.md [hunk 6]
```diff
@@ -22,7 +22,7 @@ This sample requires [Docker 17.06](https://docs.docker.com/release-notes/docker
 
 ## Prerequisites
 
-The [.NET Core 2.2 SDK](https://dotnet.microsoft.com/download) or later is required for some of the instructions in this document.
+The current [.NET SDK](https://dotnet.microsoft.com/download).
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-https.md:25]: @mthalman  please review

Comment (mthalman) [reply aspnetcore/security/docker-https.md:25]: looks good

File: aspnetcore/security/docker-https.md [hunk 7]
```diff
@@ -52,22 +52,22 @@ Use the following instructions for your operating system configuration.
...
 
-In the preceding commands, replace `<password>` with a password.
+In the preceding commands, replace `<CREDENTIAL_PLACEHOLDER>` with a password.
 
 Run the container image with ASP.NET Core configured for HTTPS in a command shell:
 
 ```console
-docker pull mcr.microsoft.com/dotnet/samples:aspnetapp
-docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:/https/ mcr.microsoft.com/dotnet/samples:aspnetapp
+docker pull mcr.microsoft.com/dotnet/core/samples:aspnetapp
```
Comment (mthalman) [inline aspnetcore/security/docker-https.md:64]: There are merge issues with the removal of `core` in the repo path.
`dotnet/core/samples` should all be changed to `dotnet/samples`

Comment (Rick-Anderson) [reply aspnetcore/security/docker-https.md]: > There are merge issues with the removal of `core` in the repo path. `dotnet/core/samples` should all be changed to `dotnet/samples`

Fixed with [this commit](https://github.com/dotnet/AspNetCore.Docs/pull/27186/commits/e45fae27ec27d13b81511e9d9022db9a063eee9a)

File: aspnetcore/security/docker-https.md [hunk 8]
```diff
@@ -76,41 +76,43 @@ Note: The certificate in this case must be a `.pfx` file.  Utilizing a `.crt` or
...
+In the preceding commands, replace `<CREDENTIAL_PLACEHOLDER>` with a password. When using [PowerShell](/powershell/scripting/overview), replace `%USERPROFILE%` with `$env:USERPROFILE`.
 
 Run the container image with ASP.NET Core configured for HTTPS:
 
 ```console
 docker pull mcr.microsoft.com/dotnet/samples:aspnetapp
-docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ --user ContainerAdministrator mcr.microsoft.com/dotnet/samples:aspnetapp
+docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="<CREDENTIAL_PLACEHOLDER>" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/samples:aspnetapp
```
Comment (mthalman) [inline aspnetcore/security/docker-https.md:111]:
```suggestion
docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="<CREDENTIAL_PLACEHOLDER>" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ --user ContainerAdministrator mcr.microsoft.com/dotnet/samples:aspnetapp
```

File: aspnetcore/security/docker-https.md [hunk 9]
```diff
@@ -76,41 +76,43 @@ Note: The certificate in this case must be a `.pfx` file.  Utilizing a `.crt` or
...
 Run the container image with ASP.NET Core configured for HTTPS:
 
 ```console
-docker pull mcr.microsoft.com/dotnet/samples:aspnetapp
-docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ --user ContainerAdministrator mcr.microsoft.com/dotnet/samples:aspnetapp
+docker pull mcr.microsoft.com/dotnet/core/samples:aspnetapp
+docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="CREDENTIAL_PLACEHOLDER" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ mcr.microsoft.com/dotnet/core/samples:aspnetapp
 ```
 
+***NOTE:*** `CREDENTIAL_PLACEHOLDER`is a placeholder for the Kestrel certificates default password.
```
[the remark this answers is not in the document]
Comment (mthalman) [reply aspnetcore/security/docker-https.md:114]: This doesn't look resolved.

File: aspnetcore/security/docker-https.md [hunk 10]
```diff
@@ -76,41 +76,43 @@ Note: The certificate in this case must be a `.pfx` file.  Utilizing a `.crt` or
...
 
 Run the container image with ASP.NET Core configured for HTTPS:
 
 ```console
 docker pull mcr.microsoft.com/dotnet/samples:aspnetapp
-docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="password" -e ASPNETCORE_Kestrel__Certificates__Default__Path=\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ --user ContainerAdministrator mcr.microsoft.com/dotnet/samples:aspnetapp
+docker run --rm -it -p 8000:80 -p 8001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=8001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="<CREDENTIAL_PLACEHOLDER>" -e ASPNETCORE_Kestrel__Certificates__Default__Path=c:\https\aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:C:\https\ --user ContainerAdministrator mcr.microsoft.com/dotnet/samples:aspnetapp
 ```
 
+***NOTE:*** `CREDENTIAL_PLACEHOLDER`is a placeholder for the Kestrel certificates default password.
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-https.md:114]:
```suggestion
***NOTE:*** `<CREDENTIAL_PLACEHOLDER>` is a placeholder for the Kestrel certificates default password.
```

**Review by mthalman — APPROVED** [review]

**Pull request merged**